tensorflow/core/kernels/mkl_concat_op.cc

```diff
@@ -715,8 +787,13 @@ class MklConcatOp : public OpKernel {
 
       if (!inputs.empty()) {
         if (are_all_mkl_inputs) {
+#ifdef ENABLE_MKLDNN_V1
+          auto concat_pd =
+              concat::primitive_desc(concat_dim, srcs_pd, cpu_engine);
+#else
           auto concat_pd = concat::primitive_desc(concat_dim, srcs_pd);
-          auto dst_pd = concat_pd.dst_primitive_desc();
+#endif  // ENABLE_MKLDNN_V1
```

Comment:
Technically, we could use
```c++
auto concat_pd = concat::primitive_desc(concat_dim, MEMORY_PD_WITHOUT_DATA(srcs_pd, cpu_engine));
```
as well. But please feel free to leave this unchanged if you think it's too ugly.

If you change it here, please also change lines 381-385 too.

Reply:
i will check and get back

Reply:
Able to the make the above recommended change in 790 to 795 but we cant do the similar change in 381-385 because two different variable required in caching srcs_pd and srcs_md

Reply:
Thank you for the explanation!